source/common/upstream/health_discovery_service.cc

```diff
@@ -203,6 +205,11 @@ void HdsDelegate::processMessage(
       cluster_config.add_health_checks()->MergeFrom(health_check);
     }
 
+    // Add transport_socket_match to cluster for use in host connections.
+    for (auto& transport_socket_match : cluster_health_check.transport_socket_matches()) {
```

Comment:
Nit: `const auto&`

Reply:
Went with the change mentioned below, about using MergeFrom directly on this field.